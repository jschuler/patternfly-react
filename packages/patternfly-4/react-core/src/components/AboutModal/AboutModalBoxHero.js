import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/AboutModalBox/styles.css';

const propTypes = {
  /** additional classes added to the About Modal Hero */
  className: PropTypes.string,
  /** the URL of the image for the Hero. */
  src: PropTypes.string,
  /** the alternate text of the Hero image. */
  alt: PropTypes.string.isRequired
};

const defaultProps = {
  className: '',
  src: ''
};

const AboutModalBoxHero = ({ className, src, alt, ...props }) => (
  <div className={css(styles.aboutModalBoxHero, className)}>
    <img {...props} className={css(styles.aboutModalBoxHeroImage)} src={src} alt={alt} />
  </div>
);

AboutModalBoxHero.propTypes = propTypes;
AboutModalBoxHero.defaultProps = defaultProps;

export default AboutModalBoxHero;
