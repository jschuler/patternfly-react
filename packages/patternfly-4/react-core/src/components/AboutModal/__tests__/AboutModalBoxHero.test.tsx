// Auto-generated do not edit

import * as React from 'react';
import { mount } from 'enzyme';
import { AboutModalBoxHero } from '../AboutModalBoxHero';

describe('AboutModalBoxHero test', () => {
  it('AboutModalBoxHero should match snapshot', () => {
    const view = mount(
      <AboutModalBoxHero
        className={undefined}
				backgroundImageSrc={undefined}
      />);
    expect(view).toMatchSnapshot();
  });
});
