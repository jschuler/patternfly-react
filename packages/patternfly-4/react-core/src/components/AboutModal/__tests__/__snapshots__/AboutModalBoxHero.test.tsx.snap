// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutModalBoxHero test AboutModalBoxHero should match snapshot 1`] = `
<Component>
  <div
    className="pf-c-about-modal-box__hero"
    style={
      Object {
        "--pf-c-about-modal-box__hero--sm--BackgroundImage": "url(undefined)",
      }
    }
  />
</Component>
`;
